src/psyclone/psyGen.py

```diff
@@ -3309,24 +3322,39 @@ def gen_code(self, parent):
         :type parent: :py:class:`psyclone.f2pygen.SubroutineGen`
 
         '''
+        def is_unit_literal(expr):
+            ''' Check if the given expression is equal to the literal '1'.
+
+            :param expr: a PSyIR expression.
+            :type expr: :py:class:`psyclone.psyGen.Node`
+
+            :return: True if it is equal to the literal '1', false otherwise.
+            '''
+            return isinstance(expr, Literal) and expr.value == '1'
         if not self.is_openmp_parallel():
             calls = self.reductions()
             zero_reduction_variables(calls, parent)
 
-        if self.root.opencl or (self.start_expr == Literal("1") and
-                                self.stop_expr == Literal("1")):
+        if self.root.opencl or (is_unit_literal(self.start_expr) and
+                                is_unit_literal(self.stop_expr)):
             # no need for a loop
             for child in self.loop_body:
                 child.gen_code(parent)
         else:
             from psyclone.psyir.backend.fortran import FortranWriter
             from psyclone.f2pygen import DoGen, DeclGen
-            # Start_expr and stop_expr are generated with the FortranWriter
-            # backend, the rest of the loop with fpygen.
+            # start/stop/step_expr are generated with the FortranWriter
+            # backend, the rest of the loop with f2pygen.
             fwriter = FortranWriter()
+            if is_unit_literal(self.step_expr):
+                step_str = None
+            else:
+                import pdb; peb.set_trace()
```

Comment:
You still have an import of pdb here which must also mean that this else clause is not covered by the test suite.

Reply:
Ups, I pushed before committing this bit, this is fixed now. Also added a test for the else condition.